Cross-compilation fails on musl targets (sched_param struct mismatch)

## Problem

Cross-compilation for `aarch64-unknown-linux-musl` and `armv7-unknown-linux-musleabihf` fails with:

```
error[E0063]: missing fields `sched_ss_init_budget`, `sched_ss_low_priority`, `sched_ss_max_repl` and 1 other field in initializer of `sched_param`
```

The musl libc's `sched_param` struct has additional POSIX fields that glibc doesn't expose. Code that initializes `sched_param` with struct literal syntax works on glibc targets but fails on musl.

## How found

Restored CI job `cross-check` (PR #167) runs `cross check` for 8 targets including two musl targets.

## Fix

Use `std::mem::zeroed()` for `sched_param` initialization instead of struct literals, or gate the musl-incompatible code behind `#[cfg(target_env = "gnu")]`.

--- a/binaries/cli/src/command/daemon.rs
+++ b/binaries/cli/src/command/daemon.rs
@@ -98,7 +98,12 @@ impl Executable for Daemon {
                 // Set SCHED_FIFO priority 50 (Linux only).
                 #[cfg(target_os = "linux")]
                 {
-                    let param = libc::sched_param { sched_priority: 50 };
+                    // Use zeroed() + field set instead of struct literal
+                    // because musl libc's sched_param has extra POSIX
+                    // fields (sched_ss_*) that glibc doesn't expose
+                    // (dora-rs/adora#170).
+                    let mut param: libc::sched_param = unsafe { std::mem::zeroed() };
+                    param.sched_priority = 50;
                     let sched_result =
                         unsafe { libc::sched_setscheduler(0, libc::SCHED_FIFO, &param) };
                     if sched_result == 0 {

